### PR TITLE
feat(Dockerfile): Update from ubuntu-18.04 to ubuntu-22.04

### LINKED ISSUE #2855: Update docker images to ubuntu-22.04
### Describe the Enhancement:
Static Analysis of vulnerabilities like [clair](https://github.com/quay/clair) detects vulnerabilities that although ubuntu-18.04 was updated, have not been resolved.

The same scan for ubuntu-22.04 ends up completely clean.

### Describe the Need:
You need to update the base image used in docker, to the latest LTS version of ubuntu, to get the latest updates related to all base software.

All users whose corporate policy includes the use of pipelines that include static analysis will benefit.

### Current Alternative
The only alternative for now is not to use the official **chef/chefworkstation** image and use the Dockerfile as a base to build your own version.

### PR DESCRIPTION
**Pull request by rnt**
Signed-off-by: Renato Covarrubias <rnt@rnt.cl>

Update Dockerfile from ubuntu-18.04 to ubuntu-22.04

## Description
Update Dockerfile from ubuntu-18.04 to ubuntu-22.04 to resolve #2855

## Related Issue
#2855

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Review by vkarve-chef — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -21,7 +21,7 @@ ENV LANG=en_US.UTF-8 \
 RUN apt-get update && \
     apt-get install -y gcc git graphviz make rsync ssh vim-tiny wget && \
     ln -s /usr/bin/vi /usr/bin/vim && \
-    wget --content-disposition "http://packages.chef.io/files/${CHANNEL}/chef-workstation/${VERSION}/ubuntu/18.04/chef-workstation_${VERSION}-1_amd64.deb" -O /tmp/chef-workstation.deb && \
+    wget --content-disposition "http://packages.chef.io/files/${CHANNEL}/chef-workstation/${VERSION}/ubuntu/22.04/chef-workstation_${VERSION}-1_amd64.deb" -O /tmp/chef-workstation.deb && \
```
Comment (vkarve-chef) [inline Dockerfile:24]: the URL where package is maintained continues to be 18.04. We are yet to shift to building package on newer Ubuntu version. this line change would need to be discarded.

**Review by vkarve-chef — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (rnt) [reply Dockerfile]: Change in ba09e683

Comment (rnt) [reply Dockerfile]: @vkarve-chef changed in 2947b0bd

**Review by vkarve-chef — APPROVED** [review]

**Comment by varnav** [comment]
You still use 18.04 package, please see #3284

**Pull request merged**